addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+        :param account.move.line line:
+        :param float net_amount:
+        :rtype: dict[str, list[dict]]
+        """
+        cls_vals = {'ecls': [], 'icls': []}
+
+        if line.l10n_gr_edi_cls_category:
+            cls_vals_list = cls_vals['ecls'] if line.l10n_gr_edi_cls_category in CLASSIFICATION_CATEGORY_EXPENSE else cls_vals['icls']
+            cls_type = line.l10n_gr_edi_cls_type or ''
+            if len(cls_type) > 0 and cls_type[0] == 'X':  # handle duplicate E3 type on inv type 17.5
+                cls_type = cls_type[1:]
+
+            cls_vals_list.append({
+                'category': line.l10n_gr_edi_cls_category,
+                'type': cls_type,
+                'amount': net_amount,
+            })
+            if line.l10n_gr_edi_cls_vat:
+                cls_vals_list.append({
+                    'category': '',
+                    'type': line.l10n_gr_edi_cls_vat,
+                    'amount': net_amount,
+                })
+
+        return cls_vals
+
+    @api.model
+    def _l10n_gr_edi_get_sum_classification_vals(self, details):
+        """
+        Aggregates the amounts from all common classifications from the ``details`` list vals.
+        :param list[dict] details:
+        :rtype: dict[str, list[dict]]
+        """
+        icls_vals, ecls_vals = {}, {}
+        summary_icls, summary_ecls = [], []
+
+        for detail in details:
+            icls_list, ecls_list = detail['icls'], detail['ecls']
+            for icls in icls_list:
+                category_type = (icls['category'], icls['type'])
+                icls_vals.setdefault(category_type, 0)
+                icls_vals[category_type] += icls['amount']
+            for ecls in ecls_list:
+                category_type = (ecls['category'], ecls['type'])
+                ecls_vals.setdefault(category_type, 0)
+                ecls_vals[category_type] += ecls['amount']
+
+        for category_type, total_amount in icls_vals.items():
+            category, cls_type = category_type
+            summary_icls.append({'type': cls_type, 'category': category, 'amount': total_amount})
+        for category_type, total_amount in ecls_vals.items():
+            category, cls_type = category_type
+            summary_ecls.append({'type': cls_type, 'category': category, 'amount': total_amount})
+
+        return {'summary_icls': summary_icls, 'summary_ecls': summary_ecls}
+
+    def _l10n_gr_edi_get_invoices_xml_vals(self):
+        """
+        Generates a dictionary containing the values needed for rendering ``l10n_gr_edi.mydata_invoice`` XML.
+        :return: dict
+        """
+        xml_vals = {'invoices': []}
+
+        for move in self.sorted(key='id'):
+            details = []
+            base_lines, _tax_lines = move._get_rounded_base_and_tax_lines()
+
+            for base_line in base_lines:
+                line = base_line['record']
+                net_amount = base_line['tax_details']['total_excluded']
```

Comment:
should probably be
```suggestion
                net_amount = base_line['tax_details']['total_excluded'] + base_line['tax_details']['delta_total_excluded']
```
or

```suggestion
                net_amount = base_line['tax_details']['raw_total_excluded']
```